tutorials/haskell/snap-web-app/tutorial.md

```diff
@@ -0,0 +1,482 @@
+                     , snap-core                 >= 0.9     && < 0.10
+                     , snap-extras               >= 0.11    && < 0.12
+                     , snap-server               >= 0.9     && < 0.10
+                     , snap-loader-static        >= 0.9     && < 0.10
+                     , text                      >= 0.11    && < 1.3
+                     , xmlhtml                   >= 0.1     && < 0.3
+                     , HDBC                      >= 2.3     && <= 2.5
+                     , HDBC-sqlite3              == 2.3.*
+                     , interpolate               >= 0.1 && < 0.2
+```
+
+Those are the required libraries for the application to be built. Most of them
+belongs to Snap's core like [`snap`](http://hackage.haskell.org/package/snap), [`snap-core`](http://hackage.haskell.org/package/snap-core) ,
+[`snap-server`](http://hackage.haskell.org/package/snap-server)
+and [`heist`](http://hackage.haskell.org/package/heist). Been this last the preferred template library by Snap's creators. There are
+others libraries like [`HDBC`](http://hackage.haskell.org/package/HDBC) and
+[`HDBC-sqlite3`](http://hackage.haskell.org/package/HDBC-sqlite3), which enable
+interaction with our database, and some others like [`text`](http://hackage.haskell.org/package/text)
+and [`interpolate`](http://hackage.haskell.org/package/interpolate) to
+allow us inmutable unicode text manipulation and strings interpolation.
+
+## Coding our way out to the browser
+
+The heart of the Snap's snaplets infrastructure is state management. Based on that
+a good start for this application will be defining a data structure to hold the state,
+that includes the state of all snaplets used by our application in a module
+called [Application](https://github.com/stackbuilders/tutorials/tutorials/haskell/snap-web-app/code/src/Todo/Application.hs).
+
+```haskell
+data App = App
+  { _heist :: Snaplet (Heist App)
+  , _conn  :: Connection
+  }
+
+```
+This data structure contains a `_heist` field holding the
+[Heist Snaplet](http://snapforbeginners.com/chapters/heist.html)  and a `_conn`
+for the database connection. This is the right place to include all the Snaplets you
+want if you're aiming to build a bigger app. Also, this data structure will be used
+later to initialize the web app in the
+[Site module](https://github.com/stackbuilders/tutorials/tutorials/haskell/snap-web-app/code/src/Todo/Site.hs#L139-L146)
+like follows:
+
+```haskell
+app :: SnapletInit App App
+app = makeSnaplet "app" "A todo application." Nothing $ do
+  h <- nestSnaplet "heist" heist $ heistInit "templates"
+  c <- liftIO $ Db.startConnection "todos.db"
+  liftIO $ Db.checkMigrationOrMigrate c
+
+  addRoutes routes
+  return $ App h c
+```
+All snaplet initializers must be wrapped in a call to [`makeSnaplet`](http://hackage.haskell.org/package/snap-0.14.0.6/docs/Snap-Snaplet.html#v:makeSnaplet),
+that's why this `app` function uses it.
```

Comment:
@nebtrx I think a period or a dash would work here instead of a comma ("... to `makeSnaplet`. That's why...").